protocols/rendezvous/src/server.rs

```diff
@@ -168,44 +167,63 @@ impl NetworkBehaviour for Behaviour {
         }
 
         let poll_res = self.inner.poll(cx, params);
-        if let Poll::Ready(ToSwarm::GenerateEvent(event)) = poll_res {
-            let opt_event = match event {
-                libp2p_request_response::Event::Message {
-                    peer: peer_id,
-                    message:
-                        libp2p_request_response::Message::Request {
-                            request, channel, ..
-                        },
-                } => {
-                    let (event, response) =
-                        handle_request(peer_id, request, &mut self.registrations);
-                    if let Some(resp) = response {
-                        self.inner
-                            .send_response(channel, resp)
-                            .expect("Send response");
-                    }
+        if let Poll::Ready(to_swarm) = poll_res {
+            match to_swarm {
+                ToSwarm::GenerateEvent(event) => {
+                    let opt_event = match event {
```

Comment:
Same as for the client, I'd prefer if we collapse these pattern matches. It drastically reduces the level of indentation which I find much easier to read.